### ISSUE TITLE
Debugger reports errors when running script with improperly encoded characters

### ISSUE THREAD

**Issue by artisticcheese** [PowerShell/vscode-powershell#868]
### System Details

- Operating system name and version:
- VS Code version: 1.13.0
- PowerShell extension version: 1.3.2 
- Output from `$PSVersionTable`:
```
Name                           Value                                                                                                                                
----                           -----                                                                                                                                
PSVersion                      5.1.15063.296                                                                                                                        
PSEdition                      Desktop                                                                                                                              
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                              
BuildVersion                   10.0.15063.296                                                                                                                       
CLRVersion                     4.0.30319.42000                                                                                                                      
WSManStackVersion              3.0                                                                                                                                  
PSRemotingProtocolVersion      2.3                                                                                                                                  
SerializationVersion           1.1.0.1                                                                                                                              
```

Copy / paste the below commands into the PowerShell Integrated Terminal, and paste the output here

code -v
$pseditor.EditorServicesVersion
code --list-extensions --show-versions
$PSVersionTable
```
PS C:\gd\Documents\s\docker\tools\vstsagent> code -v
1.13.0
376c52b955428d205459bea6619fc161fc8faacf
PS C:\gd\Documents\s\docker\tools\vstsagent> $pseditor.EditorServicesVersion

Major  Minor  Build  Revision
-----  -----  -----  --------
1      3      2      0


PS C:\gd\Documents\s\docker\tools\vstsagent> code --list-extensions --show-versions
DotJoshJohnson.xml@1.8.0
mkloubert.vs-deploy@9.25.0
ms-vscode.csharp@1.10.0
ms-vscode.PowerShell@1.3.2
PeterJausovec.vscode-docker@0.0.16
```
### Issue Description

Configuration below will throw powershell engine off and will report errors below 

```
Configuration test {
    Script DockerConfigFile {
        TestScript = {
            return $true;
        }
        GetScript = {
          
            return {@{configValue = "dd"}
            }
        }
        SetScript = {
            $dockerConfig | Add-Member –MemberType NoteProperty –Name graph –Value $graphLocation
               
        }
    }
}
Test

```

Errors reported during run

```
At C:\gd\Documents\s\docker\tools\vstsagent\test.ps1:12 char:82
+ ... ember â€“MemberTypeNoteProperty â€“Name graph â€“Value $graphLocation
+                                                     ~~~~~~~~~~~~~~~~~~~~~
The string is missing the terminator: ".

At C:\gd\Documents\s\docker\tools\vstsagent\test.ps1:11 char:21
+         SetScript = {
+                     ~
Missing closing '}' in statement block or type definition.

At C:\gd\Documents\s\docker\tools\vstsagent\test.ps1:17 char:5
+ Test
+     ~
There is an incomplete property assignment block in the instance definition.

At C:\gd\Documents\s\docker\tools\vstsagent\test.ps1:1 char:20
+ Configuration test {
+                    ~
Missing closing '}' in statement block or type definition.
```

**Comment by daviwil** [comment]
How are you running the configuration, with F8?  What is your F8 key bound to?  Open `File -> Preferences -> Keyboard Shortcuts` and search for F8.  If you see it bound to `workbench.action.terminal.runSelectedText`, remove that binding.  If not, we can continue debugging the issue.

**Comment by artisticcheese** [comment]
Running with F5

**Comment by daviwil** [comment]
Does this code work in the PowerShell console?  There shouldn't be anything special about this code that should cause the PowerShell extension to have an issue.

Might be helpful if you captured logs and posted them here so I can see if there are any other internal errors: https://github.com/PowerShell/vscode-powershell#reporting-problems

**Comment by artisticcheese** [comment]
Works fine in ISE. 
![image](https://user-images.githubusercontent.com/13176884/27056966-c2449cf8-4f8f-11e7-8af7-5ea55d3388e6.png)


[1497303932-c32fd6c2-b7c9-4468-aa4b-59a2eb9463a21497303929626.zip](https://github.com/PowerShell/vscode-powershell/files/1069670/1497303932-c32fd6c2-b7c9-4468-aa4b-59a2eb9463a21497303929626.zip)

**Comment by daviwil** [comment]
I have a suspicion that it has something to do with the file encoding.  For this file, what  encoding does VS Code report in the bottom right hand corner of the window:

![image](https://user-images.githubusercontent.com/79405/27057084-6e8d7d38-4f7f-11e7-9154-c48b3aa7e639.png)

**Comment by artisticcheese** [comment]
Says UTF-8

![image](https://user-images.githubusercontent.com/13176884/27057151-81247422-4f90-11e7-902c-23af32eeebc8.png)

**Comment by daviwil** [comment]
Ok, I'll see if I can figure out what's going on.  Thanks for the help!

**Comment by artisticcheese** [comment]
Saving as Windows 1252 does in fact work.

**Comment by Eickhel** [comment]
I actually have the same issue... kinda simpler:

```
function Convert-Exceptions {
    param ([String]$src = [String]::Empty)
    $src -replace "el medano", "el médano"
}

Convert-Exceptions("el medano")
```

Running with F8 works... result:
`el médano`
but running it F5... result:
`el mÃ©dano`

The issue gets resolved as well when setting the encoding to Windows 1252.

**Comment by tylerl0706** [comment]
Thanks for the update, @Eickhel. Feel free to 👍 the issue which will help us prioritize.

**Comment by tylerl0706** [comment]
@Eickhel are you on Windows?

I actually can't repro this on macOS.

**Comment by tylerl0706** [comment]
Ah. I've got a repro!

It's only on Windows PowerShell and this is, in fact, a Windows PowerShell bug. Here's why:

let's say we have @Eickhel's example:
```
function Convert-Exceptions {
    param ([String]$src = [String]::Empty)
    $src -replace "el medano", "el médano"
}

Convert-Exceptions("el medano")
```

With the PowerShell ISE, when you have an untitled file and hit F5, it does the same thing as if you selected all the text and hit F8 (aka ctrl-A, F8).
![image](https://user-images.githubusercontent.com/2644648/35528821-ed02014c-04e3-11e8-83c8-84096523f9ad.png)

If you save the file to something, like `test.ps1` it runs the actual file like so:
![image](https://user-images.githubusercontent.com/2644648/35528867-1b44280a-04e4-11e8-8e7c-8cca82dff6d3.png)

Notice the bad formatting!

VSCode runs the file that you are F5'ing as well:
```
PS C:\Users\tylerleonhardt\Desktop> c:\Users\tylerleonhardt\Desktop\test.ps1
```
If we run that exact line in Windows PowerShell as well we get the same wrong formatting:
![image](https://user-images.githubusercontent.com/2644648/35528634-527bc1a8-04e3-11e8-9cb1-e36867f341b3.png)

If we try this is PowerShell Core 6, the bug was fixed and the output is what we expect:
![image](https://user-images.githubusercontent.com/2644648/35528697-8f140bfc-04e3-11e8-96b8-67a85be908fc.png)

Moral of the story, this is a bug in Windows PowerShell and not the PowerShell extension for VSCode.
